File: HelloJUnit/src/com/example/junit/GreetingImplTest.java
package com.example.junit;

import static org.junit.Assert.*;

import org.junit.Test;

public class GreetingImplTest {

	@Test
	public void greetShouldReturnAValidValue() {
		//fail("Not yet implemented");
		Greeting greeting = new GreetingImpl();
		String result = greeting.greet("junit");
		assertNotNull(result);
		assertEquals("Hello junit", result);
	}

}

File: HelloMSTest/HelloTest/GreetingImplTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using HelloDotnet;

namespace HelloTest
{
    [TestClass]
    public class GreetingImplTest
    {
        [TestMethod]
        public void greetShouldReturnAValidValue()
        {
            GreetingImpl greeting = new GreetingImpl();
            Assert.IsNotNull(greeting);

            string result = greeting.greet("ABC");
            Assert.IsNotNull(result);
            Assert.AreEqual("Hello ABC", result);
        }
    }
}
